multiplexer/abci/multiplexer.go

```diff
@@ -421,6 +421,12 @@ func openTraceWriter(traceWriterFile string) (w io.WriteCloser, err error) {
 func (m *Multiplexer) getApp() (servertypes.ABCI, error) {
 	m.mu.Lock()
 	defer m.mu.Unlock()
+	return m.unsafeGetApp()
+}
+
+// unsafeGetApp gets or creates an application for the current app version. It is unsafe due to not
+// holding the mutex.
+func (m *Multiplexer) unsafeGetApp() (servertypes.ABCI, error) {
```

Comment:
Is it possible to refactor the mutex handling so that there isn't an `unsafeGetApp()`?

I don't understand when `unsafeGetApp()` should be used vs the presumably safe `getApp()`.

Reply:
scratch that, this mutex fix is required, it was just trying to avoid footguns in the future.

I'll remove for now